### PR TITLE
Add env vars for enabling socks5 and http proxies

### LINKED ISSUE #4985: FR: Expose --socks5-server , --outbound-http-proxy-listen flags in run.sh used in entrypoint for Docker container
### What are you trying to do?

In https://tailscale.com/kb/1112/userspace-networking/ ,
these two tailscaled flags are documented:

--socks5-server
--outbound-http-proxy-listen

It would be nice if these flags could be exposed in the run.sh used as the entrypoint for the tailscale docker container.

### How should we solve this?

This would allow using tailscale over a socks5 or http proxy , in more restrictive container environments
where creating a tun device is not possible.

### What is the impact of not solving this?

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by rodrigc**
- TSD_SOCKS5_SERVER, argument passed to tailscaled --socks5-server
- TSD_HTTP_PROXY, argument passed to tailscaled -outbound-http-proxy-listen
- TSD_EXTRA_ARGS extra arguments passed to tailscaled

Fixes #4985

Signed-off-by: Craig Rodrigues <rodrigc@crodrigues.org>

### CODE REVIEW

**Comment by rodrigc** [comment]
@maisem do you want me to change the variable to what you recommended here: https://github.com/tailscale/tailscale/issues/4985#issuecomment-1170793241
`TS_TAILSCALED_EXTRA_ARGS`

and get rid of the variables for SOCKS5 and HTTP_PROXY?

File: docs/k8s/run.sh [hunk 1]
```diff
@@ -15,6 +15,10 @@ TS_STATE_DIR="${TS_STATE_DIR:-}"
 TS_ACCEPT_DNS="${TS_ACCEPT_DNS:-false}"
 TS_KUBE_SECRET="${TS_KUBE_SECRET:-tailscale}"
 
+TSD_SOCKS5_SERVER="${TSD_SOCKS5_SERVER:-}"
+TSD_HTTP_PROXY="${TSD_HTTP_PROXY:-}"
```
Comment (maisem) [inline docs/k8s/run.sh:19]: TS_OUTBOUND_HTTP_PROXY_LISTEN

File: docs/k8s/run.sh [hunk 2]
```diff
@@ -15,6 +15,10 @@ TS_STATE_DIR="${TS_STATE_DIR:-}"
...
+TSD_EXTRA_ARGS="${TSD_EXTRA_ARGS:-}"
```
[the remark this answers is not in the document]
Comment (rodrigc) [reply docs/k8s/run.sh]: @maisem OK, I changed these variables to:

```
TS_SOCKS5_SERVER
TS_OUTBOUND_HTTP_PROXY_LISTEN
TS_TAILSCALED_EXTRA_ARGS
```

Fun with variable names. ;)

[the remark this answers is not in the document]
Comment (maisem) [reply docs/k8s/run.sh]: This is fine, thanks!

**Comment by maisem** [comment]
Can you make your commit message match our existing style?

(git log to see, but it's basically https://golang.org/wiki/CommitMessage)

Over here it'll be something like
`docs/k8s: add env variable for tailscaled args`

File: docs/k8s/run.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (rodrigc) [reply docs/k8s/run.sh]: @maisem OK, thanks !

**Comment by danthegoodman1** [comment]
Came here to check if this existed before I started writing code... glad I did!

**Comment by rodrigc** [comment]
@maisem OK, I changed the commit message

**Review by maisem — APPROVED** [review]

**Comment by rodrigc** [comment]
@maisem One comment about this script.  I did not want to do unrelated changes, but I observed patterns like this:

```
 if [[ ! -z "${TS_DEST_IP}" ]]; then
```

that usage is correct, but can alternatively be written as:

```
 if [[ -n "${TS_DEST_IP}" ]]; then
```

This usage can be simplified even further:

```
 if [[  "${TS_DEST_IP}" ]]; then
```

As long as your variable is enclosed in quotaiton marks this will work.

Food for thought in future patches. :)

**Pull request merged**